developers/release.md

```diff
@@ -0,0 +1,214 @@
+---
+layout: documentation
+title: How to Release Katello
+sidebar: sidebars/developer_guide.html
+---
+
+# How to Release Katello
+
+This guide covers how to peform a community release of Katello. The guide will reference the term 'X.Y' which reprsents the major and minor version that you are releasing. Throughout the guide we will use 2.0 and 2.1 as examples to represent the currently released community version (2.0) and the version that you are working on releasing (2.1).
+
+### Step 0: Code Name
+
+Pick a release code name based on a style of beer previously unused. Examples from previous releases are Oktoberfest and Winter Warmer.
+
+### Step 1: Warn Community of Branching
+
+The first step to any release is warning the community of impending branching prior to the branch date to give them time to push on any pull requests that are pending. The email should be sent no less than 2 weeks prior to the intended branch date and ideally when Foreman sends their branching announcement.
+
+### Step 2: Stable Pulp Build
+
+Prior to branching the Katello repositories, the release nanny should figure out what the current stable version of Pulp is and if not already tagged and built, do so. Please see the [Upgrading Pulp Guide](http://www.katello.org/developers/upgrade_pulp.html).
+
+### Step 3: Setup Your Environment
+
+There are a number of repositories that will need to be tagged and branched throughout the process. We recommend organizing this before hand so that you have a common place to reference and work through these tasks within throughout the release process. Clone the following:
+
+```
+git clone https://github.com/Katello/katello
+git clone https://github.com/Katello/katello-installer.git
+git clone https://github.com/Katello/katello-misc
+git clone https://github.com/Katello/katello-agent
+git clone https://github.com/Katello/hammer-cli-katello
+```
+
+Note that some repositories contain multiple spec-files and thus packages:
+
+```
+katello
+  rubygem-katello.spec
+  deploy/katello.spec
+
+katello-misc
+  certs-tools/katello-certs-tools.spec
+  repos/katello-repos.spec
+  utils/katello-utils.spec
+
+katello-agent
+  katello-agent.spec
+
+katello-installer
+  katello-installer.spec
+
+hammer-cli-katello
+  rubygem-hammer_cli_katello.spec
+```
+
+### Step 4: Configure Koji
+
+In preparation for the build, Koji needs to have configuration added specific to the tags that the release will occur under. The first step is to add new build roots and tags:
+
+```
+wave hands, perform magic
+```
```

Comment:
update to mention https://github.com/Katello/katello-misc/blob/master/rel-eng/koji_create_buildroots.sh